### PR TITLE
make blocking subdomains configurable

### LINKED ISSUE #6601: Host-based subdomain blocking breaks search functionality of AWS console
**Version info**:
```
qutebrowser v2.3.0
Git commit: 
Backend: QtWebEngine 5.15.5, based on Chromium 87.0.4280.144
Qt: 5.15.2

CPython: 3.9.6
PyQt: 5.15.4

sip: 6.1.0.dev2104271705
colorama: 0.4.4
jinja2: 3.0.1
pygments: 2.9.0
yaml: 5.4.1
adblock: no
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.4
PyQt5.QtWebKitWidgets: yes
pdf.js: 2.9.359 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.36.0
QtNetwork SSL: OpenSSL 1.1.1k  25 Mar 2021

Style: QFusionStyle
Platform plugin: wayland
OpenGL: AMD, 4.6 (Compatibility Profile) Mesa 21.1.4
Platform: Linux-5.12.15-arch1-1-x86_64-with-glibc2.33, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.9/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
```

**Does the bug happen if you start with `--temp-basedir`?**:
Yes, but _only if I run `:adblock-update` before attempting to reproduce_.

**Description**
In the [Amazon Web Services web console](https://console.aws.amazon.com/), there is a search bar at the top of the page that allows you to search AWS services and documentation. Up until recently, this search bar has worked okay. However, on updating to qutebrowser v2.3.0, it stopped working. I believe it is linked specifically to this change:

> The hosts-based adblocker (using `content.blocking.hosts.lists`) now also blocks all requests to any subdomains of blocked hosts.

Disabling adblocking with `:set content.blocking.enabled false` allows the search bar to work again.

Using dev tools, you can see requests being blocked to https://concierge.analytics.console.aws.a2z.com/search. The blocklist at https://github.com/StevenBlack/hosts contains an entry for analytics.console.aws.a2z.com, but _not_ for concierge.analytics.console.aws.a2z.com, which leads me to conclude that the blocked functionality is a result of the aforementioned change.

Perhaps a toggle to turn this subdomain blocking functionality off would be appropriate? I did not notice such a parameter mentioned in the changelog or when looking at parameters under `content.blocking`.

**How to reproduce**
1. Run `qutebrowser --temp-basedir`.
2. Update your adblock list using `:adblock-update`.
3. Navigate to https://console.aws.amazon.com/
4. Log in with a valid AWS account. Creating an AWS account is free.
5. In the search bar at the top of the page, attempt to search (any search string is OK). Notice the message "Unified search service is unavailable. Try your search again later."
6. Run `:set content.blocking.enabled false`.
7. Refresh the page.
8. In the search bar at the top, attempt to search (any search string is OK). Notice that the search functionality now works.

### PR DESCRIPTION
**Pull request by 4nd3r**
fixes #6601

### CODE REVIEW

File: tests/unit/components/test_hostblock.py [hunk 1]
```diff
@@ -570,4 +570,7 @@ def test_subdomain_blocking(config_stub, host_blocker_factory):
     config_stub.val.content.blocking.hosts.lists = None
     host_blocker = host_blocker_factory()
     host_blocker._blocked_hosts.add("example.com")
+    config_stub.val.content.blocking.hosts.block_subdomains = True
```
Comment (The-Compiler) [inline tests/unit/components/test_hostblock.py:573]: Please use [parametrization](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) to turn this into two different test cases (and move this line to the other settings above).

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Thanks! LGTM overall, just a small suggestion to improve the test.

File: tests/unit/components/test_hostblock.py (hunk 1 above)
Comment (4nd3r) [reply tests/unit/components/test_hostblock.py]: TIL. might be bit hackish, but I added comment about.

**Comment by The-Compiler** [comment]
Thanks! Merged with some small doc changes in a9e4aeeb4.

**Pull request merged**